src/components/views/ScreeningsCard.vue

```diff
@@ -51,12 +56,25 @@ export default defineComponent({
     getScreeningTime(screening) {
       return new Date(screening.datetime).toLocaleString('en-GB', {timeStyle: 'short'})
     },
+    async fetchMovieData() {
+      try {
+        const response = await this.moviesStore.fetchMoviesToState()
+        console.log(response)
+      } catch(error) {
+        console.error(error)
+      } finally {
+        this.isRequestFinished = true
+      }
+    },
+  },
+  beforeMount() {
+    this.fetchMovieData()
```

Comment:
don't you get movie data from props?  if so - it probably shouldn't be the responsibility of this component to load it 🤔 also you are getting all movies (I guess) even though this component is just for one movie - you will probably make a lot of the same requests if there will be a couple of `ScreeningsCard` on the one page

Reply:
The problem occurs in the moviesStore. When I refresh the ChooseSeatsPage, the movies array in the store get empty so the ScreeningCard has no data to be generated and the whole component fails to render. Probably it would be better to prevent the store clearing than to fetch all data again -- would you have some hint for me on how to achieve that?

Reply:
Ultimately I have tested all subpages if they render when accessed by providing a url and I have added independent requests where needed. We have discussed this question during the summary meeting and were advised by Szymon to make requests always, even if the data is accessible, to be sure we don't use anything that is out of date.